x-pack/plugins/security_solution/public/cloud_security_posture/components/entity_insight.tsx

```diff
@@ -123,7 +79,7 @@ export const EntityInsight = <T,>({
         <VulnerabilitiesPreview
           name={name}
           isPreviewMode={isPreviewMode}
-          hasNonClosedAlerts={alertsCount > 0}
+          hasNonClosedAlerts={hasNonClosedAlerts}
```

Comment:
Let's try to get rid of passing `hasNonClosedAlerts` to both `VulnerabilitiesPreview` and `MisconfigurationsPreview`. As you moved the logic to `use_non_closed_alerts` you can get this information from inside the components, or even better from `use_entity_insight` and `use_risk_score_data`. In the end you pass it here only to pass it down to these hooks